pkg/tableformatter/tableformatter.go

```diff
@@ -0,0 +1,298 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tableformatter
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+)
+
+type TableFormatter[T any] struct {
+	options            TableFormatterOptions
+	columns            []*ColumnInfo[T]
+	showColumns        []*ColumnInfo[T]
+	columnMap          map[string]*ColumnInfo[T]
+	fillString         string
+	currentScreenWidth int
+}
+
+// NewTableFormatter creates a new TableFormatter for type T
+func NewTableFormatter[T any](options *TableFormatterOptions) (*TableFormatter[T], error) {
+	if options == nil {
+		options = GetDefaultOptions()
+	}
+
+	entryPrototype := new(T)
+
+	t := reflect.TypeOf(entryPrototype)
+	if t.Kind() == reflect.Pointer {
+		t = t.Elem()
+	}
+
+	tf := &TableFormatter[T]{
+		columns:   make([]*ColumnInfo[T], 0),
+		columnMap: make(map[string]*ColumnInfo[T]),
+		options:   *options,
+	}
+
+	err := tf.iterateFields(t, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Register default columns (where Visible == true)
+	tf.SetShowDefaultColumns()
+
+	return tf, nil
+}
+
+func MustNewTableFormatter[T any](options *TableFormatterOptions) *TableFormatter[T] {
+	tf, err := NewTableFormatter[T](options)
+	if err != nil {
+		panic(err)
+	}
+	return tf
+}
```

Comment:
I renamed it to MustCreateTableFormatter() for now. I like to use this pattern when initializing things globally. The regex package uses it as well, for example. The advantage of using it that way is that the program crashes instantly and makes you aware of any syntactical mistakes immediately instead of failing (maybe even silently) only when certain code paths are run.